katdal/test/test_chunkstore_s3.py

```diff
@@ -305,9 +331,39 @@ def do_all(self):
             'upgrade', 'proxy-authorization', 'proxy-authenticate'
         }
         self.protocol_version = 'HTTP/1.1'
-        url = urllib.parse.urljoin(self.server.target, self.path)
         data_len = int(self.headers.get('Content-Length', 0))
         data = self.rfile.read(data_len)
+        truncate = False
+
+        # Extract a proxy suggestion prepended to the path
+        suggestion = re.search(r'/please-([^/]+?)(?:-for-([\d\.]+)-seconds)*/',
+                               self.path)
```

Comment:
I started with `re.match` and then moved on to `re.search` because it turned out to be very useful to have fault injections in the middle of `path` (otherwise it replaces the bucket name, which causes carnage with token authorisation).

Reply:
I think that's why I chose to have the redundant `please` in the suggestion (besides general kindness) - no test should have such a silly term in its paths :-) So I don't think more is needed (hopefully tests break anyway).